spec/spec.md

```diff
@@ -1597,6 +1597,81 @@ JSONPath                      | Description
+not allow for usage of Regular Expressions in a normative way.
+
+#### JSONPath
+
+JSONPath is being used in this specification both in the Input Descriptors of the Presentation Definition,
+defining where certain data is to be found, and the Presentation Submission,
+defining how data maps from the definition to the Verifiable Presentation.
+With Regular Expressions come certain security considerations, that must be taken into account.
+Be aware that in a future major version of the spec we envision to have some normative texts in the specification
+itself disallowing certain aspects of JSONPath altogether like function extensions (i.e., Regular Expressions)
+
+Please also read the JSONPath IETF document providing more details on some of the security considerations mentioned
+below.
+
+
+#### Attack Vectors on JSONPath Implementations
+
+Historically, JSONPath has often been implemented by feeding parts of the query to an underlying 
+programming language engine, e.g.,  JavaScript's eval() function.  This approach is well known to lead to 
+injection attacks and would require perfect input validation to prevent these attacks. 
+Since that is hard to achieve we recommend restricting JSONPath queries to not allow for functions 
+(i.e., Regular Expressions, see above)
+
+Attacks on availability may attempt to trigger unusually expensive runtime performance exhibited by certain 
+implementations in certain cases. This particularly applies to the use of Regular Expressions. We recommend restricting 
+JSONPath queries to not allow for function extensions.
+Function extensions like `match` and `search` are using regular expressions, 
+to which the above security considerations apply. 
+
+#### Impact of not using function extensions in JSONPath
+As such it is recommended that software implementors do not allow for function extensions in JSONPath and relying parties do not create definitions that
+rely on functions.
+
+Not using functions does restricts the input descriptor part in terms of flexibility. 
+On the other hand, most issuers know exactly what type of credentials and claims they want to receive, 
+so having very complex matching logic with Regular Expressions involved is not typically needed.
+
+#### Restrict submission data to exact JSONPath paths
+JSON path has filtering logic, but once you are submitting your Verifiable Presentation and submission data, you 
+should be correlating input descriptor field ids in the `descriptor_map` with exact paths to the Verifiable Credentials 
+or claims. We suggest to not allow for both filters and function extensions for parties implementing the specification.
+
+#### Attack Vectors on JSON Schema
+JSON Schema validation allows the use of Regular Expressions for instance in filters, to which the security 
+considerations, mentioned above, apply. Regular Expressions could be used in the `filter` of a `field` as part
+of a `constraint`, by using the `pattern` property. We advise not to use this property.
```

Comment:
We'll also want to exclude the `regex` format keyword https://json-schema.org/draft/2020-12/json-schema-validation.html#section-7.3.8

and `patternProperties` https://json-schema.org/draft/2020-12/json-schema-core.html#section-10.3.2.2